### PR TITLE
Added get_by_id and delete function for distributed firewall rules

### PR DESCRIPTION
**Pull request by navalkp**
Added corresponding unit tests

### CODE REVIEW

**Review by waynew — COMMENTED** [review]
I didn't notice any glaring issues, but @dhiltonp may catch something that I've missed

File: src/saltext/vmware/modules/vmc_distributed_firewall_rules.py [hunk 1]
```diff
@@ -116,3 +116,166 @@ def get(
         cert=cert,
         params=params,
     )
+
+
+def get_by_id(
```
[the remark this answers is not in the document]
Comment (shaikres) [reply src/saltext/vmware/modules/vmc_distributed_firewall_rules.py:121]: As discussed, we will take this as an improvement to move sensitive information to pillar data and do the changes in all modules and states and raise it as a separate PR.

changes will be made as part of the issue: https://github.com/saltstack/salt-ext-modules-vmware/issues/121

File: src/saltext/vmware/modules/vmc_distributed_firewall_rules.py [hunk 2]
```diff
@@ -116,3 +116,166 @@ def get(
...
+    hostname,
+    refresh_key,
```
[the remark this answers is not in the document]
Comment (shaikres) [reply src/saltext/vmware/modules/vmc_distributed_firewall_rules.py:123]: The `refresh_key`  is one time generated value which can be valid for longer time or can be a never expired token. 

please refer below:
image

image

Based on this `refresh_key` and  `authorization_host`, we will generate an access_token programatically within the function which is used for CRUD operations

File: src/saltext/vmware/modules/vmc_distributed_firewall_rules.py [hunk 3]
```diff
@@ -116,3 +116,166 @@ def get(
...
+    authorization_host,
```
[the remark this answers is not in the document]
Comment (shaikres) [reply src/saltext/vmware/modules/vmc_distributed_firewall_rules.py:124]: As discussed, we will take this as an improvement (along with moving sensitive information to pillar data) and do the changes in all modules and states and raise it as a separate PR

changes will be made as part of the issue: https://github.com/saltstack/salt-ext-modules-vmware/issues/121

File: src/saltext/vmware/modules/vmc_distributed_firewall_rules.py [hunk 4]
```diff
@@ -116,3 +116,166 @@ def get(
...
+    org_id,
+    sddc_id,
+    domain_id,
+    security_policy_id,
+    rule_id,
+    verify_ssl=True,
```
[the remark this answers is not in the document]
Comment (shaikres) [reply src/saltext/vmware/modules/vmc_distributed_firewall_rules.py:130]: we will remove the cert parameter and use only verify_ssl to handle this.
we will consider this as an improvement and do the changes in all modules and states and raise it as separate PR

File: src/saltext/vmware/modules/vmc_distributed_firewall_rules.py [hunk 5]
```diff
@@ -116,3 +116,166 @@ def get(
...
+
+
+def get_by_id(
+    hostname,
+    refresh_key,
+    authorization_host,
+    org_id,
+    sddc_id,
+    domain_id,
+    security_policy_id,
```
[the remark this answers is not in the document]
Comment (navalkp) [reply src/saltext/vmware/modules/vmc_distributed_firewall_rules.py:128]: Its user defined and few pre created security policies are there its string. https://developer.vmware.com/docs/nsx-vmc-policy/latest/policy/api/v1/infra/domains/domain-id/security-policies/security-policy-id/rules/rule-id/get/

File: src/saltext/vmware/modules/vmc_distributed_firewall_rules.py [hunk 6]
```diff
@@ -116,3 +116,166 @@ def get(
...
+        The Id of SDDC for which the distributed firewall rule should be retrieved
+
+    domain_id
+        The domain_id for which the distributed firewall rule should be retrieved
+
+    security_policy_id
+        Security policy id for which rule should be retrieved
+
+    rule_id
+        Id of the distribute firewall rule to be retrieved from SDDC
```
[the remark this answers is not in the document]
Comment (navalkp) [reply src/saltext/vmware/modules/vmc_distributed_firewall_rules.py]: updated this doc.

**Review by dhiltonp — APPROVED** [review]

File: src/saltext/vmware/modules/vmc_distributed_firewall_rules.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dhiltonp) [reply src/saltext/vmware/modules/vmc_distributed_firewall_rules.py:128]: I wish this were in the docs but it's probably ok.

**Pull request merged**